### PR TITLE
Fix example code in the `SpecificMatcher` cop

### PR DESCRIPTION
**Pull request by boris-petrov**
The title says it all.

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `main` (if not - rebase it).
- [x] Squashed related commits together.
- [ ] Added tests.
- [x] Updated documentation.
- [ ] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: lib/rubocop/cop/capybara/specific_matcher.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ module Capybara
       #   expect(page).to have_no_link('foo', class: 'cls', href: 'http://example.com')
       #   expect(page).to have_table(class: 'cls')
       #   expect(page).to have_select
-      #   expect(page).to have_field('foo')
+      #   expect(page).to have_field(with: 'foo')
```
Comment (ydah) [inline lib/rubocop/cop/capybara/specific_matcher.rb:26]: question) Example to verify that the name attribute has an input field of 'foo', is there a problem with the code in the original example?

Comment (boris-petrov) [reply lib/rubocop/cop/capybara/specific_matcher.rb:26]: It's not exactly a problem per se, it's just that all the rest of the example correspond to a line in the [bad](https://github.com/rubocop/rubocop-capybara/blob/8b63fc80d26639fac81eb26d9e6cd692adbe81b0/lib/rubocop/cop/capybara/specific_matcher.rb#L10) part. Only the last one is different - and I think the intention was for it also to be the same - just the correct way to write it. Adding the `with:` does that - makes it with the same meaning as the "bad" example.

**Comment by ydah** [comment]
Could you check failed CI?

**Comment by boris-petrov** [comment]
Sorry about that. I think I fixed it.

**Review by ydah — APPROVED** [review]
Thank you!

**Pull request merged**